R/npn_data_download.R

```diff
@@ -719,10 +731,22 @@ npn_get_data <- function(url,
                                promote_num_to_string = TRUE) %>%
       tibble::as_tibble() %>%
       #replace missing data indicator with NA
-      dplyr::mutate(dplyr::across(dplyr::where(is.numeric),
-                                  \(x) ifelse(x == -9999, NA_real_, x))) %>%
-      dplyr::mutate(dplyr::across(dplyr::where(is.character),
-                                  \(x) ifelse(x == "-9999", NA_character_, x)))
+      dplyr::mutate(
+        dplyr::across(dplyr::where(is.numeric),
+                      function(x) ifelse(x == -9999, NA_real_, x))
+      ) %>%
+      dplyr::mutate(
+        dplyr::across(dplyr::where(is.character),
+                      function(x) ifelse(x == "-9999", NA_character_, x))
+      ) %>%
+      #handle some columns that may be read in as the wrong type if they happen
+      #to be all NAs (#87). `any_of()` is used because not all endpoints will
+      #return these columns!
+      dplyr::mutate(
+        dplyr::across(dplyr::any_of("update_datetime"),
+                      function(x) as.POSIXct(x, tz = "UTC")),
```

Comment:
Not sure what the timezone should be here or if it matters.  Could also just make this column character and leave it up to the user to figure out the timezone if they care.